threescale_api/resources.py

```diff
@@ -762,6 +762,19 @@ def oidc(self):
     def backend_usages(self) -> 'BackendUsages':
         return BackendUsages(parent=self, instance_klass=BackendUsage)
 
+    @property
+    def active_docs(self) -> 'ActiveDocs':
+        """ Active docs related to service. """
+        up_self = self
+
+        class Wrap(ActiveDocs):
+            def list(self, **kwargs) -> List['DefaultResource']:
```

Comment:
why this implementation? a) I see it select(s)_by b) and also already heard from smadis. How about update of ActiveDocs.list or ActiveDocs.update for easier reusability? e.g. ActiveDocs.list(service_id=None)

Reply:
@mganisin I think it is logical to get activedocs related to service/product from service/product without using extra params(like you've mentioned). Usage of `service['id']` or `service.entity_id` should be encapsulated in the `service.activedocs.list()` call.
`_service_instance_.active_docs` is similar to all other related objects, like `_service_instance_.metrics`. This and my implementation return object where you can call `list`. So both are not evaluated when you run the getter `_service_instance_.active_docs` but at time you call `list()` on that getter. I've tried to simulate the same behavior like for other properties.
Current implementation is last from couple of them and it seems to me the best solution, don't you think?